syntax = "proto3";

package proto.ext.clipboard;

message ClipboardHtmlContent {
  string html = 1;
  string text = 2;
};

message ClipboardPathContent {
  string path = 1;
};

message ClipboardOptions {
  bool concealed = 1;
};

message CopyToClipboardRequest {
  ClipboardContent content = 1;
  ClipboardOptions options = 2;
};

message ClipboardContent {
  oneof content {
    string text = 1;
    ClipboardHtmlContent html = 2;
    ClipboardPathContent path = 3;
  };
};

message CopyToClipboardResponse {
};

message Request {
  oneof payload {
    CopyToClipboardRequest copy = 1;
  };
};

message Response {
  oneof payload {
    CopyToClipboardResponse copy = 1;
  };
};
